File: Bai_131/main.cpp
#include<bits/stdc++.h>
using namespace std;

using ll = long long;
string ktn(ll n, int k, int t) {
    ll m = n;
    string kq = "";
    while (t != k) {
        int h = n % 10;
        n /= 10;
        if (h != 5) 
            t += 1;
        kq = "5" + kq;
    }
    if (stoll(to_string(n) + kq) < m) 
        n += 1;
    return to_string(n) + kq;
}

string ktl(ll n, int k, int t) {
    string num = to_string(n);
    t = 0;
    for (int i = 0; i < num.length(); i++) {
        if (num[i] == '5') 
            t += 1;
        if (t == k + 1) 
            return num.substr(0, i) + "6" + string(num.length() - i - 1, '0');
    }
    return num;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    freopen("MIN.INP", "r", stdin);
    freopen("MIN.OUT", "w", stdout);
    ll n; int k; cin >> n >> k;
    int t = 0;
    for (char c : to_string(n)) 
        if (c == '5') 
            t += 1;
    if (t < k) cout << ktn(n, k, t);
    else cout << ktl(n, k, t);
    return 0;
}

File: Bai_131/main.py
def ktn(n, k, t):
    m = n
    kq = ""
    while t != k:
        h = n % 10
        n //= 10
        if h != 5:
            t += 1
        kq = "5" + kq
    if int(str(n) + kq) < m:
        n += 1
    return str(n) + kq

def ktl(n, k, t):
    n = str(n)
    t = 0
    for i in range(len(n)):
        if n[i] == "5":
            t += 1
        if t == k + 1:
            return n[:i] + "6" + "0" * (len(n) - i - 1)
        
with open("MIN.INP", "r") as ifile:
    n, k=map(int, ifile.read().split())

t = str(n).count("5")

with open("MIN.OUT", "w") as ofile:
    if t < k: ofile.write(ktn(n,k,t))
    else: ofile.write(ktl(n,k,t))